### PR TITLE
Update CASCADE + CHECK constraints

### LINKED ISSUE #2506: sql: Prevent adding both a cascading action and a check on the same column
PR: https://github.com/cockroachdb/cockroach/pull/21690

From release notes:
> Prevented adding both a cascading referential constraint action and a check constraint to a column.

### PR DESCRIPTION
**Pull request by lhirata**
Closes #2506.

### CODE REVIEW

File: v2.0/foreign-key.md [hunk 1]
```diff
@@ -58,9 +58,9 @@ Parameter | Description
...
-`ON DELETE SET NULL` / `ON UPDATE SET NULL` | When a referenced foreign key is deleted or updated, respectively, the columns of all rows referencing that key will be set to `NULL`. The column must allow `NULL` or this update will fail.
-`ON DELETE SET DEFAULT` / `ON UPDATE SET DEFAULT` | When a referenced foreign key is deleted or updated, respectively, the columns of all rows referencing that key are set to the default value for that column. If the default value for the column is null, this will have the same effect as `ON DELETE SET NULL` or `ON UPDATE SET NULL`. The default value must still conform with all other constraints, such as `UNIQUE`.
+`ON DELETE CASCADE` / `ON UPDATE CASCADE` | When a referenced foreign key is deleted or updated, all rows referencing that key are deleted or updated, respectively. If there are other alterations to the row, such as a `SET NULL` or `SET DEFAULT`, the delete will take precedence. `CASCADE` cannot be added to a column that has a [`CHECK` constraint](check.html). <br><br>Note that `CASCADE` does not list objects it drops or updates, so it should be used cautiously.
```
Comment (jseldess) [inline v2.0/foreign-key.md]: I think this would sit better in the next paragraph with the note.

File: v2.0/foreign-key.md [hunk 2]
```diff
@@ -58,9 +58,9 @@ Parameter | Description
...
+`ON DELETE SET NULL` / `ON UPDATE SET NULL` | When a referenced foreign key is deleted or updated, respectively, the columns of all rows referencing that key will be set to `NULL`. The column must allow `NULL` or this update will fail. `SET NULL` cannot be added to a column that has a [`CHECK` constraint](check.html).
```
Comment (jseldess) [inline v2.0/foreign-key.md]: I think this would sit better as a separate paragraph, perhaps beginning "Note that..."

File: v2.0/foreign-key.md [hunk 3]
```diff
@@ -58,9 +58,9 @@ Parameter | Description
...
+`ON DELETE SET DEFAULT` / `ON UPDATE SET DEFAULT` | When a referenced foreign key is deleted or updated, respectively, the columns of all rows referencing that key are set to the default value for that column. If the default value for the column is null, this will have the same effect as `ON DELETE SET NULL` or `ON UPDATE SET NULL`. The default value must still conform with all other constraints, such as `UNIQUE`. `SET DEFAULT` cannot be added to a column that has a [`CHECK` constraint](check.html).
```
Comment (jseldess) [inline v2.0/foreign-key.md]: Same as above.

**Comment by lhirata** [comment]
Review status: 0 of 2 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

*[v2.0/foreign-key.md, line 61 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2780#-L8XskVFDkTtDfaoLu59-r1-61:-L8Yuqn8DgX9ujq9ojck:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/aa2325dea31af719fdd0974e7b28f6d92809a585/v2.0/foreign-key.md#L61)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

I think this would sit better in the next paragraph with the note.
</blockquote></details>

Done.

---

*[v2.0/foreign-key.md, line 62 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2780#-L8XskVFDkTtDfaoLu59-r1-62:-L8YutQ4pW_8u0MCS947:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/aa2325dea31af719fdd0974e7b28f6d92809a585/v2.0/foreign-key.md#L62)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

I think this would sit better as a separate paragraph, perhaps beginning "Note that..."
</blockquote></details>

Done.

---

*[v2.0/foreign-key.md, line 63 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2780#-L8XskVFDkTtDfaoLu59-r1-63:-L8Yuvj4sMMjl63NMQnJ:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/aa2325dea31af719fdd0974e7b28f6d92809a585/v2.0/foreign-key.md#L63)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Same as above.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/2780)*

**Comment by BramGruneir** [comment]
So actually, this isn't a limitation anymore.  They should work together now.

I do think there is a still a limitation with computed columns.

**Comment by lhirata** [comment]
Looks like it's not a limitation with computed columns either. I was just able to create the following table:
```
CREATE TABLE abc (
    a INT PRIMARY KEY,
    b INT,
    c INT,
    d INT CHECK (d > 0) AS (a + b + c) STORED
  );
```

Is that correct or am I doing something wrong?

---

Review status: 0 of 2 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2780#-:-L8cq15bDbOFHwV-JS1s:b-qgc3kj)*

**Comment by lhirata** [comment]
Oh wait, you mean cascade + computed columns?

---

Review status: 0 of 2 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2780#-:-L8crEfp1VvtQoo988RN:b-f3pdwb)*

**Comment by BramGruneir** [comment]
Yep, a mix of cascade an computed columns.

---

Review status: 0 of 2 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2780#-:-L8csv640czPgAAU4Hul:b-clsv3k)*

**Comment by BramGruneir** [comment]
:lgtm:

---

Review status: 0 of 2 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2780#-:-L8d-GiSEah4kTYMzfZm:bnfp4nl)*

**Comment by lhirata** [comment]
TFTRs! 🎉

**Pull request merged**